src/main/kotlin/andrasferenczi/templater/constructorTemplate.kt

```diff
@@ -36,8 +36,8 @@ fun createConstructorTemplate(
                     addNewLine()
 
                     publicVariableNames.forEach {
-                        if (addRequiredAnnotation) {
-                            addTextSegment("@required")
+                        if (addRequiredAnnotation || it.isFinal) {
```

Comment:
Could you explain a bit why only the `final` fields get to have `required`? In terms of null-safety it looks like it is needed in both `private` and `public` fields, not sure why you would differentiate between them.

Reply:
> Could you explain a bit why only the `final` fields get to have `required`? In terms of null-safety it looks like it is needed in both `private` and `public` fields, not sure why you would differentiate between them.

I'm very sorry to late answer @andrasferenczi 

Your are right.
All fields except `nullable` need to be `required`.

I hope @mzdm create new PR applied this opinion.

Reply:
no issue! thanks for starting this PR